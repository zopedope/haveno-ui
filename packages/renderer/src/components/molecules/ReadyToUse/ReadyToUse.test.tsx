// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import { ReadyToUse } from ".";

describe("molecules::ReadyToUse", () => {
  it("renders without exploding", () => {
    const spy = vi.fn();
    const { asFragment, unmount } = render(<ReadyToUse onSubmit={spy} />);
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("calls onSubmit", () => {
    const spy = vi.fn();
    const { unmount } = render(<ReadyToUse onSubmit={spy} />);
    expect(spy).to.not.toHaveBeenCalled();
    fireEvent.click(screen.getByRole("button"));
    expect(spy).to.toHaveBeenCalledTimes(1);
    unmount();
  });
});
