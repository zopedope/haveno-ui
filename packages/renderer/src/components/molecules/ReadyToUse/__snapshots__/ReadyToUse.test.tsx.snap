// Vitest Snapshot v1

exports[`molecules::ReadyToUse > renders without exploding 1`] = `
<DocumentFragment>
  <div
    class="mantine-Stack-root mantine-lfk3cq"
  >
    <h1
      class="mantine-Title-root mantine-1krktmh"
    >
      Haveno is ready for use.
    </h1>
    <div
      class="mantine-Text-root mantine-8at25l"
    >
      Youâ€™ve succesfully set up Haveno. Please note that to be able to trade, you need to deposit Monero in your Haveno wallet and set up a payment account.
    </div>
    <div
      class="mantine-63n06h"
    />
    <div
      class="mantine-Group-root mantine-19jxmdp"
    >
      <button
        class="mantine-Button-filled mantine-Button-root mantine-Group-child mantine-1hjgevl"
        type="submit"
      >
        <div
          class="mantine-3xbgk5 mantine-Button-inner"
        >
          <span
            class="mantine-qo1k2 mantine-Button-label"
          >
            Start using Haveno
          </span>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
